flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/UnboundedNonPartitionedProcessingOverProcessFunction.scala

```diff
@@ -0,0 +1,107 @@
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.runtime.state.{FunctionInitializationContext, FunctionSnapshotContext}
+import org.apache.flink.streaming.api.checkpoint.CheckpointedFunction
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.apache.flink.table.functions.{Accumulator, AggregateFunction}
+import org.apache.flink.types.Row
+import org.apache.flink.util.{Collector, Preconditions}
+
+/**
+  * Process Function used for the aggregate in
+  * [[org.apache.flink.streaming.api.datastream.DataStream]]
+  *
+  * @param aggregates the list of all [[org.apache.flink.table.functions.AggregateFunction]]
+  *                   used for this aggregation
+  * @param aggFields  the position (in the input Row) of the input value for each aggregate
+  */
+class UnboundedNonPartitionedProcessingOverProcessFunction(
+    private val aggregates: Array[AggregateFunction[_]],
+    private val aggFields: Array[Int],
+    private val forwardedFieldCount: Int,
+    private val aggregationStateType: RowTypeInfo)
+  extends ProcessFunction[Row, Row] with CheckpointedFunction{
+
+  Preconditions.checkNotNull(aggregates)
+  Preconditions.checkNotNull(aggFields)
+  Preconditions.checkArgument(aggregates.length == aggFields.length)
+
+  private var accumulators: Row = _
+  private var output: Row = _
+  private var state: ListState[Row] = null
+
+  override def open(config: Configuration) {
+    output = new Row(forwardedFieldCount + aggregates.length)
+  }
+
+  override def processElement(
+    input: Row,
+    ctx: ProcessFunction[Row, Row]#Context,
+    out: Collector[Row]): Unit = {
+
+    if (null == accumulators) {
+      val it = state.get().iterator()
+      if (it.hasNext) {
+        accumulators = it.next()
+      } else {
+        accumulators = new Row(aggregates.length)
+        var i = 0
+        while (i < aggregates.length) {
+          accumulators.setField(i, aggregates(i).createAccumulator())
+          i += 1
+        }
+      }
+    }
+
+    var i = 0
+    while (i < forwardedFieldCount) {
+      output.setField(i, input.getField(i))
+      i += 1
+    }
+
+    i = 0
+    while (i < aggregates.length) {
+      val index = forwardedFieldCount + i
+      val accumulator = accumulators.getField(i).asInstanceOf[Accumulator]
+      aggregates(i).accumulate(accumulator, input.getField(aggFields(i)))
+      output.setField(index, aggregates(i).getValue(accumulator))
+      i += 1
+    }
+
+    out.collect(output)
+  }
+
+  override def snapshotState(context: FunctionSnapshotContext): Unit = {
+    state.clear()
+    if (null != accumulators) {
+      state.add(accumulators)
+    }
+  }
+
+  override def initializeState(context: FunctionInitializationContext): Unit = {
+    val stateSerializer =
+      aggregationStateType.createSerializer(getRuntimeContext.getExecutionConfig)
+    val accumulatorsDescriptor = new ListStateDescriptor[Row]("overState", stateSerializer)
```

Comment:
When you review FLINK-5803 you had told me that use `TypeInformation` instead of `TypeSerializer`，I had tried this, but unfortunately reported exception. So, I use TypeSerializer to create `ListStateDescriptor`. the exception info：
```
Caused by: java.lang.IllegalStateException: Serializer not yet initialized.
	at org.apache.flink.api.common.state.StateDescriptor.getSerializer(StateDescriptor.java:169)
	at org.apache.flink.api.common.state.ListStateDescriptor.getElementSerializer(ListStateDescriptor.java:93)
	at org.apache.flink.runtime.state.DefaultOperatorStateBackend.getOperatorState(DefaultOperatorStateBackend.java:110)
	at org.apache.flink.runtime.state.DefaultOperatorStateBackend.getOperatorState(DefaultOperatorStateBackend.java:91)
	at org.apache.flink.table.runtime.aggregate.UnboundedNonPartitionedProcessingOverProcessFunction.initializeState(UnboundedNonPartitionedProcessingOverProcessFunction.scala:104)
	at org.apache.flink.streaming.api.functions.util.StreamingFunctionUtils.tryRestoreFunction(StreamingFunctionUtils.java:178)
```
Do you want me fix this, and use `aggregationStateType`, If so, I glad to try.